frontend/src/main/components/School/SchoolForm.js

```diff
@@ -56,16 +57,16 @@ function SchoolForm({ initialContents, submitAction, buttonLabel = "Create" }) {
                 </Col>
                 <Col>
                     <Form.Group className="mb-3" >
-                        <Form.Label htmlFor="termRegex">Term Regex</Form.Label>
+                        <Form.Label htmlFor="termRegex">Term Regex(Example: W24)</Form.Label>
                         <Form.Control
                             data-testid="SchoolForm-termRegex"
                             id="termRegex"
                             type="text"
                             isInvalid={Boolean(errors.termRegex)}
-                            {...register("termRegex", { required: true })}
+                            {...register("termRegex", { required: true, pattern: term_regex })}
```

Comment:
```suggestion
                            {...register("termRegex", { required: true})}
```

This is applying the regular expression `const term_regex = /[WSMF]\d\d/;` to the input for this field, which completely misunderstands what is supposed to be happening here.

Instead, we want the user to *input the regular expression against which the terms for their school will be matched*.

That is, if your schools has terms likes F24, W25, S25, M25 then your regex is `[FMSW]\d\d`.  And it should be used to validate the input of the term on the *course entry form*, not anything else.

But, we probably cannot come up with a "regex for regexes" so instead, we leave this required but we don't validate it with a regex.